timesheets/Code.gs

```diff
@@ -0,0 +1,149 @@
+  sheet.insertColumnAfter(COLUMN_NUMBER.HOURLY_PAY);
+  var weeklyPayColumn = sheet.getRange(1, COLUMN_NUMBER.CALC_PAY).setValue('WEEKLY PAY');
+  
+  // adds new approval column
+  sheet.insertColumnAfter(COLUMN_NUMBER.CALC_PAY);
+  var approvalColumn = sheet.getRange(1, COLUMN_NUMBER.APPROVAL).setValue('APPROVAL');
+
+  // make sure approval column is all drop-down menus
+  var bigRange = sheet.getRange(frozenRows + 1, COLUMN_NUMBER.APPROVAL, lastRow - frozenRows, 1);  
+  var dropdownValues = ['APPROVED', 'NOT APPROVED', 'IN PROGRESS'];
+  var rule = SpreadsheetApp.newDataValidation().requireValueInList(dropdownValues).build();
+  bigRange.setDataValidation(rule);
+  bigRange.setValue('IN PROGRESS');
+  
+  // adds new notified column
+  sheet.insertColumnAfter(COLUMN_NUMBER.APPROVAL); //global
+  var notifiedColumn = sheet.getRange(1, COLUMN_NUMBER.APPROVAL + 1).setValue('NOTIFIED STATUS'); 
+  
+  // make sure notified column is all drop-down menus
+  bigRange = sheet.getRange(frozenRows + 1, COLUMN_NUMBER.APPROVAL + 1, lastRow - frozenRows, 1);
+  dropdownValues = ['NOTIFIED', 'NOT NOTIFIED'];
+  rule = SpreadsheetApp.newDataValidation().requireValueInList(dropdownValues).build();
+  bigRange.setDataValidation(rule);
+  bigRange.setValue('NOT NOTIFIED');
+}
+
+/**
+* Runs the "Calculate Pay" script, creating a new column with each employee's 
+* pay based on their reported hours and hourly wage.
+*/
+function calculatePay() {
+  // declares variables
+  var sheet = SpreadsheetApp.getActiveSheet();
+  var lastRow = sheet.getLastRow();
+  var lastCol = sheet.getLastColumn();
+  var frozenRows = sheet.getFrozenRows();
+ 
+  // gets hours & hourly rate values
+  var rangeValues = sheet.getRange(frozenRows + 1, COLUMN_NUMBER.HOURS_START, lastRow - frozenRows, 6).getValues();
+  var pays = [];
+  
+  // goes through every employee
+  for (var i = 0; i <= lastRow - (frozenRows + 1); i++) {
+    var numHours = 0;
+    for (var j = 0; j < 5; j++) {
+      numHours += rangeValues[i][j];
+    
+    // pushes weekly pay value into 2D array
+    pays.push([numHours * sheet.getRange(i + (frozenRows + 1), COLUMN_NUMBER.HOURLY_PAY).getValue()]);
+  
+  // pushes entire 2D array of pay values into column 
+  var pushRange = sheet.getRange(frozenRows + 1, COLUMN_NUMBER.CALC_PAY, lastRow - 1, 1).setValues(pays);
+}
+
+/**
+* Runs the script "Send Emails". Sends e mails to every employee notifying 
+* them whether or not their timesheet has been approved.
+* Checks + updates "NOTIFIED" status so employers do not send e mails more than once.
+*/
+function sendEmails() {
+    var sheet = SpreadsheetApp.getActiveSheet();
+    var lastRow = sheet.getLastRow();
+    var lastCol = sheet.getLastColumn();
+    // lastCol here is the NOTIFIED column
+    var frozenRows = sheet.getFrozenRows();
+  
+    // gets ranges of values all at once
+    var emailValues = sheet.getRange(frozenRows + 1, COLUMN_NUMBER.EMAIL, lastRow - frozenRows, 1).getValues();
```

Comment:
Maybe calculate `lastRow - frozenRows` once, and use it on all three of these lines.